packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -2107,51 +2147,57 @@ export class MergeTree {
 			this.slideAckedRemovedSegmentReferences(removedSegments.map(({ segment }) => segment));
 		}
 
-		if (this.collabWindow.collaborating && seq !== UnassignedSequenceNumber) {
-			if (MergeTree.options.zamboniSegments) {
-				zamboniSegments(this);
-			}
+		if (
+			this.collabWindow.collaborating &&
+			seq !== UnassignedSequenceNumber &&
+			MergeTree.options.zamboniSegments
+		) {
+			zamboniSegments(this);
 		}
 	}
 
 	/**
 	 * Revert an unacked local op
 	 */
 	// eslint-disable-next-line import/no-deprecated
-	public rollback(op: IMergeTreeDeltaOp, localOpMetadata: SegmentGroup) {
+	public rollback(op: IMergeTreeDeltaOp, localOpMetadata: SegmentGroup): void {
 		if (op.type === MergeTreeDeltaType.REMOVE) {
 			const pendingSegmentGroup = this.pendingSegments.pop?.()?.data;
 			if (pendingSegmentGroup === undefined || pendingSegmentGroup !== localOpMetadata) {
 				throw new Error("Rollback op doesn't match last edit");
 			}
-			pendingSegmentGroup.segments.forEach((segment: ISegmentLeaf) => {
-				const segmentSegmentGroup = segment.segmentGroups?.pop?.();
+			for (const segment of pendingSegmentGroup.segments) {
+				// Cast is needed because parent was removed from ISegment
```

Comment:
Probably worth reverting this one as well, and disabling the rule, given the type differences between the patterns.